tough/src/schema/key.rs

```diff
@@ -12,71 +14,113 @@ use std::collections::HashMap;
 use std::fmt;
 use std::str::FromStr;
 
+/// Serializes signing keys as defined by the TUF specification. All keys have the format
+/// ```json
+///  { "keytype" : "KEYTYPE",
+///     "scheme" : "SCHEME",
+///     "keyval" : "KEYVAL"
+///  }
+/// ```
+/// where:
+/// ```text
+///    KEYTYPE is a string denoting a public key signature system, such
+///    as RSA or ECDSA.
+///
+///    SCHEME is a string denoting a corresponding signature scheme.  For
+///    example: "rsassa-pss-sha256" and "ecdsa-sha2-nistp256".
+///
+///    KEYVAL is a dictionary containing the public portion of the key.
+/// ```
 #[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
 #[serde(rename_all = "kebab-case")]
 #[serde(tag = "keytype")]
 pub enum Key {
+    /// An RSA key.
     Rsa {
+        /// The RSA key.
         keyval: RsaKey,
+        /// The field denoting the key type as RSA.
         scheme: RsaScheme,
-
+        /// Any additional fields.
         #[serde(flatten)]
         _extra: HashMap<String, Value>,
     },
+    /// An Ed25519 key.
     Ed25519 {
+        /// The Ed25519 key.
         keyval: Ed25519Key,
+        /// The field denoting the key type as Ed25519.
```

Comment:
```suggestion
        /// Denotes the key's signature scheme.
```

The key variant, `Ed25519`, is already named after a specific signature scheme (and, therefore, this scheme enum only has the one variant) so this comment is always going to be a bit wonky, I think...  (It makes more sense for Ecdsa keys, where there are other schemes but we only support one...)

Reply:
Yeah that works.